### ISSUE TITLE
Sync: error from number of columns mismatch does not fail the parse

### ISSUE THREAD

**Issue by q42jaap** [adaltas/node-csv-parse#151]
https://github.com/adaltas/node-csv-parse/blob/master/lib/sync.js#L26
The error returned from `__write` is ignored.

The result should be checked.

**Comment by dobesv** [comment]
Yes, I noticed the exact same issue today, test case:

```
$ node
> parse = require('csv-parse/lib/sync')
> parse('A,B\nB\nC,K', { trim: true })
[ [ 'A', 'B' ] ]
```

Should do something else, I think? I expected it to just take however many columns there were in each row.

**Comment by wdavidw** [comment]
Thanks for reporting this issue (and insisting: ), fix with 34119bd0a6058ff84e1e08e6300f5931f3cb2c6d
